ui/components/app/transaction-detail/transaction-detail.component.js

```diff
@@ -11,21 +11,28 @@ import { COLORS } from '../../../helpers/constants/design-system';
 import { PRIORITY_LEVEL_ICON_MAP } from '../../../helpers/constants/gas';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 
-export default function TransactionDetail({ rows = [], onEdit }) {
+export default function TransactionDetail({
+  rows = [],
+  onEdit,
+  userAcknowledgedGasMissing,
+}) {
   // eslint-disable-next-line prefer-destructuring
   const EIP_1559_V2 = process.env.EIP_1559_V2;
 
   const t = useI18nContext();
   const {
     gasLimit,
+    hasSimulationError,
     estimateUsed,
     maxFeePerGas,
     maxPriorityFeePerGas,
     transaction,
   } = useGasFeeContext();
   const { openModal } = useTransactionModalContext();
 
-  if (EIP_1559_V2 && estimateUsed) {
+  const editEnabled = !hasSimulationError || userAcknowledgedGasMissing;
```

Comment:
should we also check that onEdit function is present still?
```suggestion
  const editEnabled = onEdit && (!hasSimulationError || userAcknowledgedGasMissing);
```

Reply:
This check is not more needed now.

Reply:
This was hack in parent component it was not sending `onEdit` is the condition was not met `!hasSimulationError || userAcknowledgedGasMissing;`

Reply:
Why's that? If the onEdit method isn't present and we try to call it here won't it error?

Reply:
It is not called not in V2 code, only legacy code uses it and if editEnabled is. true onEdit is also present.

Reply:
I updated the PR for legacy condition